File: src/pocketcloud/cloud/network/packet/impl/normal/PlayerTransferPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl\normal;

use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\data\PacketData;
use pocketcloud\cloud\player\CloudPlayerManager;

final class PlayerTransferPacket extends CloudPacket {

    public function __construct(
        private string $player = "",
        private string $server = ""
    ) {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->write($this->player)->write($this->server);
    }

    public function decodePayload(PacketData $packetData): void {
        $this->player = $packetData->readString();
        $this->server = $packetData->readString();
    }

    public function handle(ServerClient $client): void {
        if (($player = CloudPlayerManager::getInstance()->get($this->player)) !== null) {
            $player->getCurrentProxy()?->sendPacket($this);
        }
    }

    public static function create(string $player, string $server): self {
        return new self($player, $server);
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/network/packet/impl/PlayerTransferPacket.java
package de.pocketcloud.cloud.network.packet.impl;

import de.pocketcloud.cloud.PocketCloud;
import de.pocketcloud.cloud.network.client.ServerClient;
import de.pocketcloud.network.packet.AuthenticatedPacket;
import de.pocketcloud.network.packet.ClientboundPacket;
import de.pocketcloud.network.packet.CloudboundPacket;
import de.pocketcloud.cloud.network.packet.CloudPacket;
import de.pocketcloud.network.packet.data.PacketData;
import de.pocketcloud.cloud.player.CloudPlayer;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

@NoArgsConstructor
@Getter
public final class PlayerTransferPacket extends CloudPacket implements ClientboundPacket, CloudboundPacket, AuthenticatedPacket {

    private String player;
    private String server;

    public PlayerTransferPacket(String player, String server) {
        this.player = player != null ? player : "";
        this.server = server != null ? server : "";
    }

    @Override
    public void handle(@NotNull ServerClient client) {
        PocketCloud.instance().players().get(player).flatMap(CloudPlayer::currentProxy).ifPresent(proxy -> proxy.sendPacket(this));
    }

    @Override
    public void encodePayload(PacketData packetData) {
        packetData.writeAll(player, server);
    }

    @Override
    public void decodePayload(PacketData packetData) {
        this.player = packetData.readString();
        this.server = packetData.readString();
    }

    public static PlayerTransferPacket create(String player, String server) {
        return new PlayerTransferPacket(player, server);
    }
}
